Neuro/Minigames/Solvers/SafeSolver.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections;
+using Neuro.Cursor;
+using UnityEngine;
+
+namespace Neuro.Minigames.Solvers;
+
+[MinigameSolver(typeof(SafeMinigame))]
+public sealed class SafeSolver : GeneralMinigameSolver<SafeMinigame>
+{
+    public override IEnumerator CompleteMinigame(SafeMinigame minigame, NormalPlayerTask task)
```

Comment:
This is broken. Sometimes it doesn't finish

Reply:
[`2c0b30d`](https://github.com/VedalAI/neuro-amongus/pull/76/commits/2c0b30defeb722983aa6a805e285e46bf4f638d5)
It seems to be extremely susceptible to lag at specific times, for now I've put in an error check for the tumbler combo to repeat if it failed. However, it still needs to be made more robust.

Reply:
The solver looks good, sometimes it seems to have weird edge cases where it fails though. You could try using data from the actual minigame to make sure that a certain step has been reached or something

https://clips.twitch.tv/AmericanPunchyHareTheTarFu-fvjcwCtcClMs0Avo

Reply:
> The solver looks good, sometimes it seems to have weird edge cases where it fails though. You could try using data from the actual minigame to make sure that a certain step has been reached or something

The solver is using data from the minigame to check if the tumbler is solved, however if you notice it uses `latched[1]`, not the last in the array `latched[2]`, this is because for some reason the `minigame.latched` array isn't getting updated.

In addition to that the minigame seemingly randomly unlatches the first number when switching directions which is that edge case. I've added an error check to hopefully make that edge case a bit less jarring, however because `minigame.latched` doesn't update the edge case can't be detected immediately. https://github.com/VedalAI/neuro-amongus/pull/76/commits/e3c5527bf55fa0b61fd26045e2b9116c3609a5b3

Reply:
It still occasionally messes up a bit, but it seems to recover I guess

https://clips.twitch.tv/DignifiedDepressedMeatloafImGlitch--dEWaM2gEok8Lzma